src/backup.ts

```diff
@@ -0,0 +1,492 @@
+   * const backupCompleted = (state === 'READY');
+   */
+  async getState(): Promise<
+    | EnumKey<typeof databaseAdmin.spanner.admin.database.v1.Backup.State>
+    | undefined
+  > {
+    const [backupInfo] = await this.getMetadata();
+    return backupInfo.state || undefined;
+  }
+
+  /**
+   * Retrieves the expiry time of the backup.
+   *
+   * @see {@link #updateExpireTime}
+   * @see {@link #getMetadata}
+   *
+   * @method Backup#getExpireTime
+   * @returns {Promise<PreciseDate>}
+   *     when resolved, contains the current expire time of the backup if it
+   *     exists.
+   *
+   * @example
+   * const {Spanner} = require('@google-cloud/spanner');
+   * const spanner = new Spanner();
+   * const instance = spanner.instance('my-instance');
+   * const backup = instance.backup('my-backup');
+   * const expireTime = await backup.getExpireTime();
+   * console.log(`Backup expires on ${expireTime.toISOString()}`);
+   */
+  async getExpireTime(): Promise<PreciseDate | undefined> {
+    const [backupInfo] = await this.getMetadata();
+    return new PreciseDate(backupInfo.expireTime as DateStruct);
+  }
+
+  /**
+   * Checks whether the backup exists.
+   *
+   * @see {@link #getMetadata}
+   *
+   * @method Backup#exists
+   * @returns {Promise<boolean>} when resolved, contains true if the backup
+   *     exists and false if it does not exist.
+   *
+   * @example
+   * const {Spanner} = require('@google-cloud/spanner');
+   * const spanner = new Spanner();
+   * const instance = spanner.instance('my-instance');
+   * const backup = instance.backup('my-backup');
+   * const alreadyExists = await backup.exists();
+   * console.log(`Does backup exist? ${alreadyExists}`);
+   */
+  async exists(): Promise<boolean> {
+    try {
+      // Attempt to read metadata to determine whether backup exists
+      await this.getMetadata();
+      // Found therefore it exists
+      return true;
+    } catch (err) {
+      if (err.code === status.NOT_FOUND) {
+        return false;
+      }
+      // Some other error occurred, rethrow
+      throw err;
+    }
+  }
+
+  updateExpireTime(
+    expireTime: PreciseDate
+  ): Promise<databaseAdmin.spanner.admin.database.v1.IBackup>;
+  updateExpireTime(
+    expireTime: PreciseDate,
+    gaxOptions?: CallOptions
+  ): Promise<databaseAdmin.spanner.admin.database.v1.IBackup>;
+  updateExpireTime(
+    expireTime: PreciseDate,
+    callback: UpdateExpireTimeCallback
+  ): void;
+  updateExpireTime(
+    expireTime: PreciseDate,
+    gaxOptions: CallOptions,
+    callback: UpdateExpireTimeCallback
+  ): void;
+  /**
+   * @callback UpdateExpireTimeCallback
+   * @param {?Error} err Request error, if any.
+   * @param {IBackup} backup The updated {@link v1.IBackup}.
+   */
+  /**
+   * Sets the expiry time of a backup.
+   *
+   * @see {@link #getExpireTime}
+   *
+   * @method Backup#updateExpireTime
+   * @param {PreciseDate} expireTime The expiry time to update with.
```

Comment:
That's an interesting idea. I looked at the other places where users pass in timestamps and it seems that we generally don't expect people to pass in a PreciseDate. So I changed the type of expireTime to string|number|protobuf.Timestamp. Then internally we convert it to PreciseDate.

Reply:
That sounds good, thank you. I didn't see it in the latest code, however. 🤔

Reply:
Oh right, I forgot to update this comment. I ended up not converting this one for a good reason.

`getExpireTime()` still returns a PreciseDate since I think existing methods that return timestamps also return PreciseDate. Then I thought I wanted to support the use case where you can do `updateExpireTime(getExpireTime() + <some hours/days>)`. But maybe I'm overthinking this so I'll try to change this one to accept string|number|protobuf.Timestamp.

Reply:
All done.